libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpH265Reader.java

```diff
@@ -0,0 +1,307 @@
+    trackOutput.sampleData(data, numBytesInData);
+    fragmentedSampleSizeBytes += numBytesInData;
+
+    int nalHeaderType = (data.getData()[0] >> 1) & 0x3F;
+    bufferFlags = getBufferFlagsFromNalType(nalHeaderType);
+  }
+
+  /**
+   * Processes an AP packet (RFC7798 Section 4.4.2).
+   *
+   * <p>Outputs the received aggregation units (with start code prepended) to {@link #trackOutput}.
+   * Sets {@link #bufferFlags} and {@link #fragmentedSampleSizeBytes} accordingly.
+   */
+  @RequiresNonNull("trackOutput")
+  private void processAggregationPacket(ParsableByteArray data) throws ParserException {
+    // TODO: Support AggregationPacket mode.
+    throw ParserException.createForMalformedManifest(
+        "need to implement processAggregationPacket",
+        /* cause= */ null);
+  }
+
+  /**
+   * Processes Fragmentation Unit packet (RFC7798 Section 4.4.3).
+   *
+   * <p>This method will be invoked multiple times to receive a single frame that is broken down
+   * into a series of fragmentation units in multiple RTP packets.
+   *
+   * <p>Outputs the received fragmentation units (with start code prepended) to {@link
+   * #trackOutput}. Sets {@link #bufferFlags} and {@link #fragmentedSampleSizeBytes} accordingly.
+   */
+  @RequiresNonNull("trackOutput")
+  private void processFragmentationUnitPacket(ParsableByteArray data, int packetSequenceNumber) {
+    //  The structure of an FU packet.
+    //    0                   1                   2                   3
+    //    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    //   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    //   |    PayloadHdr (Type=49)       |   FU header   | DONL (cond)   |
+    //   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-|
+    //   | DONL (cond)   |                                               |
+    //   |-+-+-+-+-+-+-+-+                                               |
+    //   |                         FU payload                            |
+    //   |                                                               |
+    //   |                               +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    //   |                               :...OPTIONAL RTP padding        |
+    //   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    //   FU header
+    //   +---------------+
+    //   |0|1|2|3|4|5|6|7|
+    //   +-+-+-+-+-+-+-+-+
+    //   |S|E|  FuType   |
+    //   +---------------+
+    //   Structure of HEVC NAL unit header
+    //   +---------------+---------------+
+    //   |0|1|2|3|4|5|6|7|0|1|2|3|4|5|6|7|
+    //   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    //   |F|   Type    |  LayerId  | TID |
+    //   +-------------+-----------------+
+
+    int tid =
+        (data.getData()[1] & 0x7); // last 3 bits in byte 1 of payload header, RFC7798 Section 1.1.4
+    int fuHeader = data.getData()[2];
+    int nalUnitType = fuHeader & 0x3F;
+    byte nalHeader[] = new byte[2];
```

Comment:
I feel we can skip creating this small array, by merging L240 and 256, L242 and 257. 

given this method is called very frequently.